src/why3task.ml

```diff
@@ -0,0 +1,41 @@
+(** Handling tasks in why3. *)
+
+open Extra
+
+(** [declare_symbols l] add the declaration of every symbol in [l] to [tsk] *)
+let declare_symbols : Why3.Term.lsymbol list -> Why3.Task.task = fun l ->
+    List.fold_left Why3.Task.add_param_decl None l
+
+(** [add_goal tsk f] add a goal with [f] formula in the task [tsk] *)
+let add_goal : Why3.Task.task -> Why3.Term.term -> Why3.Task.task =
+    fun tsk f ->
+    let new_goal = Why3.Ident.id_fresh "lambdapi_goal" in
+    let goal = Why3.Decl.create_prsymbol new_goal in
+    Why3.Task.add_prop_decl tsk Why3.Decl.Pgoal goal f
+
+(** [add_hypothesis tsk (axiom_name, axiom_term)] add the axiom [axiom_term]
+    named [axiom_name] to the Why3 task [tsk]. *)
+let add_hypothesis :
+    string -> Why3.Term.term -> Why3.Task.task ->  Why3.Task.task =
+    fun axiom_name axiom_term tsk ->
+    let new_axiom = Why3.Ident.id_fresh axiom_name in
+    let axiom = Why3.Decl.create_prsymbol new_axiom in
+    Why3.Task.add_prop_decl tsk Why3.Decl.Paxiom axiom axiom_term
+
+(** [declare_axioms l tsk] add all axioms that [l] contains to the Why3 task
+    [tsk]. *)
+let declare_axioms :
+    Why3.Term.term StrMap.t -> Why3.Task.task -> Why3.Task.task =
+    fun l tsk -> StrMap.fold add_hypothesis l tsk
```

Comment:
You can unfold declare_axioms as it is very short.

Reply:
Fixed in a29414085332857be96b62ab67a96960b46d746d